## Import NF-L KO (Beatrix) dataset
library(BiocManager)
install()
library(limma)
options(java.parameters = "-Xmx8000m")
library(XLConnect)
library(Cairo) # nicer graphics, anti-aliased, etc.
library(NMF) # this package has a great annotated heatmap function - aheatmap
library(igraph)
library(gplots)

nfl.all.btx <- loadWorkbook("~/Bioinformatics/David Proteomics/Beatrix_NFL KO_061019/ResultsSummary_10June2019_EK.xlsx") # btx ~ Beatrix
nfl.all.btx <- readWorksheet(nfl.all.btx, sheet = 2, header = T)
nfl.all.btx[1:5,1:15]
dim(nfl.all.btx) # 4739   33

## Import INA KO (Beatrix) dataset
ina.all.btx <- loadWorkbook("~/Bioinformatics/David Proteomics/INA KO 041519/INA KO_DataSummary_Beatrix_15Apr2019.xlsx") # btx ~ Beatrix
ina.all.btx <- readWorksheet(ina.all.btx, sheet = 1, header = T)
nfl.all.btx[1:5,1:15]
dim(ina.all.btx) # 3823   31

## Import TKO (Neubert) dataset
tko.all.nbt <- loadWorkbook("~/Bioinformatics/David Proteomics/NF proteomics data 9-7-17/TKO MS (14).xlsx") # nbt ~ Neubert
tko.all.nbt <- readWorksheet(tko.all.nbt, sheet = 1, header = T)
tko.all.nbt[1:5,1:9]  ## Make sure that the intensities are log10LFQ as opposed log2LFQ in Beatrix datasets
dim(tko.all.nbt) # 3953   37


## Removing unwanted columns from each data set before combining them
nfl.LFQ.btx <- nfl.all.btx[,c(1:12,22,23)]
head(nfl.LFQ.btx)
row.names(nfl.LFQ.btx) <- nfl.all.btx$Gene # Duplicate warning
sum(duplicated(nfl.all.btx$Gene)) # 27
sum(is.na(nfl.all.btx$Gene)) # 27
sum(nfl.all.btx$Gene == "") # NA

nfl.all.btx[duplicated(nfl.all.btx$Gene) | duplicated(nfl.all.btx$Gene, fromLast = T), c(1:12,23)]

nfl.LFQ.btx_uq <- nfl.all.btx_uq[!is.na(nfl.all.btx_uq$Gene), c(1:12,23)]
dim(nfl.LFQ.btx_uq) # 4711   13

sum(duplicated(nfl.LFQ.btx_uq$Gene)) # 1
nfl.LFQ.btx_uq <- nfl.all.LFQ_uq[!duplicated(nfl.LFQ.btx_uq$Gene), ]
dim(nfl.LFQ.btx_uq) # 4711   13

head(nfl.LFQ.btx_uq)
colnames(nfl.LFQ.btx_uq)[7:12] <- paste0("WT",".", paste0(rep("LKO",6),1:6))

colnames(nfl.LFQ.btx_uq)

colnames(nfl.LFQ.btx_uq)[1:12] <- paste0(colnames(nfl.LFQ.btx_uq)[1:12], ".btx")
head(nfl.LFQ.btx_uq)

row.names(nfl.LFQ.btx_uq) <- nfl.LFQ.btx_uq$Gene; nfl.LFQ.btx_uq <- nfl.LFQ.btx_uq[,-13]
head(nfl.LFQ.btx_uq)

#nfl.LFQ.btx_uq <- nfl.LFQ.btx_uq[,-13]
#head(nfl.LFQ.btx_uq)

library(mice)
md.pattern(nfl.LFQ.btx_uq)
md.pattern(nfl.LFQ.btx)
## Imputation of missing values
library(impute)
library(preprocessCore)

#nfl.LFQ.btx.impute <- impute.knn(as.matrix(nfl.LFQ.btx_uq))


ina.LFQ.btx <- ina.all.btx[,c(1:12,24,25)]
head(ina.LFQ.btx)
#row.names(ina.LFQ.btx) <- ina.all.btx$Uniprot # NO Duplicate warning
#sum(duplicated(ina.all.btx$Gene)) # 20
#sum(is.na(ina.all.btx$Gene)) # 21
#sum(ina.all.btx$Gene == "") # NA

#ina.all.btx[is.na(ina.all.btx$Gene), 1:25]
#ina.all.btx[duplicated(ina.all.btx$Gene) | duplicated(ina.all.btx$Gene, fromLast = T), 1:25]

#nfl.LFQ.btx_uq <- nfl.all.btx_uq[!is.na(nfl.all.btx_uq$Gene), c(1:12,23)]
#dim(nfl.LFQ.btx_uq) # 4712   13

#sum(duplicated(nfl.LFQ.btx_uq$Gene)) # 1
#nfl.LFQ.btx_uq <- nfl.all.LFQ_uq[!duplicated(nfl.LFQ.btx_uq$Gene), ]
#dim(nfl.LFQ.btx_uq) # 4711   13

#head(nfl.LFQ.btx)

#row.names(nfl.LFQ.btx_uq) <- nfl.LFQ.btx_uq$Gene
#head(nfl.LFQ.btx_uq)

#nfl.LFQ.btx <- nfl.LFQ.btx[,-13]
#head(nfl.LFQ.btx)

head(ina.LFQ.btx)
colnames(ina.LFQ.btx)[1:6] <- paste0("I", colnames(ina.LFQ.btx)[1:6])
colnames(ina.LFQ.btx)

colnames(ina.LFQ.btx)[7:12] <- paste0("WT",".", paste0(rep("IKO",6),1:6))
head(ina.LFQ.btx)

colnames(ina.LFQ.btx)[1:13] <- paste0(colnames(ina.LFQ.btx)[1:13], ".btx")
head(ina.LFQ.btx)



tko.LFQ.nbt <- tko.all.nbt[,c(1:8,35,37)]
head(tko.LFQ.nbt)
#row.names(nfl.LFQ.btx) <- nfl.all.btx$Uniprot 

## Convert log10 values to log2 values
tko.LFQ.nbt[,-c(9,10)] <- as.data.frame(lapply(tko.LFQ.nbt[,-c(9,10)], function(v)
  ifelse(!is.na(v), log2(10^v), v)))

head(tko.LFQ.nbt)
tail(tko.LFQ.nbt)

colnames(tko.LFQ.nbt)[1:8] <- paste0(rep(c("WT","TKO"),each=4), rep(1:4,2))
head(tko.LFQ.nbt)

colnames(tko.LFQ.nbt)[1:4] <- paste0("WT",".", paste0(rep("TKO",4),1:4))
head(tko.LFQ.nbt)

colnames(tko.LFQ.nbt)[10] <- "Gene"
colnames(tko.LFQ.nbt)[1:9] <- paste0(colnames(tko.LFQ.nbt)[1:9], ".nbt")
head(tko.LFQ.nbt)

## Merging 3 dataframes with Gene column
head(nfl.LFQ.btx)
head(ina.LFQ.btx)
head(tko.LFQ.nbt)
dim(nfl.LFQ.btx) # 4739   14
dim(ina.LFQ.btx) # 3823   14
dim(tko.LFQ.nbt) # 3953   10

df.merged <- merge(nfl.LFQ.btx, ina.LFQ.btx, by="Gene", all = T)
head(df.merged)
tail(df.merged)
dim(df.merged) # 5438   27

df.merged <- merge(df.merged, tko.LFQ.nbt, by="Gene", all = T)
head(df.merged)

run.seq <- function(x) as.numeric(ave(paste(x), x, FUN = seq_along))

L <- list(nfl.LFQ.btx, ina.LFQ.btx, tko.LFQ.nbt)
L2 <- lapply(L, function(x) cbind(x, run.seq = run.seq(x$Gene)))

df.merged <- Reduce(function(...) merge(..., all = TRUE), L2)[-2]
head(df.merged)
tail(df.merged)
dim(df.merged) # 8800   35


## create datTrait
datTraits <- loadWorkbook("datTraits_NF.xlsx")
datTraits <- readWorksheet(datTraits, sheet = 1, header = T)
datTraits






































































































































































































































































































































































































































































































































































































































































































































































































































































































































































































































































































































